### PR TITLE
Added finfo docstrings

### PR DESCRIPTION
**Pull request by rensortino**

### CODE REVIEW

**Comment by rensortino** [comment]
Hi, thank you for merging my PR. Maybe you meant " solves #2029 "?

**Comment by simonetgordon** [comment]
Yes indeed! Mistakingly referenced this PR instead. Thanks!

File: ivy/functional/ivy/data_type.py [hunk 1]
```diff
@@ -231,18 +231,83 @@ def finfo(type: Union[ivy.Dtype, str, ivy.Array, ivy.NativeArray]) -> Finfo:
...
+    
           smallest representable number.
+    
         - **smallest_normal**: *float*
+    
           smallest positive floating-point number with full precision.
 
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.data_type_functions.can_cast.html>`_ # noqa
```
Comment (Neelaksh-Singh) [inline ivy/functional/ivy/data_type.py:258]: Hi, I referred your PR for Iinfo Docstring. I believe that the api standard link for finfo is [this](https://data-apis.org/array-api/latest/API_specification/generated/signatures.data_type_functions.finfo.html)

Comment (rensortino) [reply ivy/functional/ivy/data_type.py:258]: Hi, sorry I must have left the wrong link. Thank you for letting me notice, I will change it in my next PR

Comment (Neelaksh-Singh) [reply ivy/functional/ivy/data_type.py:258]: No worries. 😊

**Pull request merged**